### PR TITLE
[IMP] util/pg: add a query format utility

### PR DESCRIPTION
**Pull request by aj-fuentes**
We want to move away of manual formatting.

### CODE REVIEW

**Comment by KangOl** [comment]
upgradeci retry with only base

File: src/util/pg.py [hunk 1]
```diff
@@ -112,6 +112,18 @@ def execute(query):
             )
 
 
+def format(cr, query, **kwargs):
+    """
+    Format the query replacing all keyword args as SQL indentifiers
+    Example:
+    ```
+    >>> util.format(cr, "SELECT id FROM {table}", table="res_users")
+    SELECT id FROM res_users
```
Comment (aj-fuentes) [inline src/util/pg.py:121]: Just to not forget updating the docstring :)
```suggestion
    SELECT id FROM "res_users"
```

**Comment by KangOl** [comment]
upgradeci retry with always only account account_followup analytic approvals base crm delivery hr hr_expense hr_holidays hr_payroll hr_recruitment lunch mrp mrp_workorder product project sale_renting sale_subscription website_slides_forum in all versions

File: src/util/pg.py [hunk 2]
```diff
@@ -629,21 +648,51 @@ def get_depending_views(cr, table, column):
...
+        head = sql.SQL(", " if self._leading_comma and self else "")
+        tail = sql.SQL("," if self._trailing_comma and self else "")
+
+        if not self._alias:
+            builder = sql.Identifier
+        elif hasattr(sql.Identifier, "strings"):
+            builder = partial(sql.Identifier, self._alias)
+        else:
+            # older psycopg2 versions, doesn't support passing multiple strings to the constructor
+            builder = lambda elem: sql.SQL(".").join(sql.Identifier(self._alias) + sql.Identifier(elem))
```
[the remark this answers is not in the document]
Comment (aj-fuentes) [reply src/util/pg.py:675]: Ah never mind `Identifier + Identifier = Composed`

**Comment by KangOl** [comment]
upgradeci retry

**Comment by KangOl** [comment]
@aj-fuentes are you confident about this one?

File: src/util/pg.py [hunk 3]
```diff
@@ -629,21 +649,76 @@ def get_depending_views(cr, table, column):
...
+
+        body = sql.SQL(", ").join(builder(elem) for elem in self._unquoted_colums)
```
Comment (aj-fuentes) [inline src/util/pg.py:703]:
```suggestion
        body = sql.SQL(", ").join(builder(elem) for elem in self._unquoted_columns)
```

Comment (aj-fuentes) [reply src/util/pg.py]: Never mind, I just pushed this as a fixup.

File: src/util/pg.py [hunk 4]
```diff
@@ -112,6 +117,20 @@ def execute(query):
             )
 
 
+def format(cr, query, *args, **kwargs):
```
Comment (KangOl) [inline src/util/pg.py:120]: As shown by the example in the docstring, `format` is too generic.
```suggestion
def format_query(cr, query, *args, **kwargs):
```

File: src/util/pg.py [hunk 5]
```diff
@@ -629,21 +648,79 @@ def get_depending_views(cr, table, column):
...
 
 
+class ColumnList(UserList, sql.Composable):
+    """
+    Encapsulate a list of elements that represent column names.
+    The resulting object can be rendered as string with leading/trailing comma or an alias.
+
+    Examples:
+    ```
+    >>> columns = ColumnList(["id", "field_Yx"])
```
Comment (KangOl) [inline src/util/pg.py:658]: Wrong API. The quoted list should also be provided.
```suggestion
    >>> columns = ColumnList(["id", "field_Yx"], ["id", '"field_Yx"'])
```

Actually, the `ColumnList` objects are not created directly but through the `util.get_columns` function or via the `using` method.

**Review by nseinlet — APPROVED** [review]

**Comment by KangOl** [comment]
@robodoo rebase-ff r+

**Pull request closed without merging**